types/number.go

```diff
@@ -86,24 +85,25 @@ func (n Number) Equal(other attr.Value) bool {
 	return n.Value.Cmp(o.Value) == 0
 }
 
+// IsNull returns true if the Number represents null value.
 func (n Number) IsNull() bool {
-	return n.Null
+	return n.Null || (!n.Unknown && n.Value == nil)
```

Comment:
I'm 👍 for this change, however it should include a changelog entry, maybe something along the lines of an enhancement of types: The Number type IsNull method will now return true for the zero-value of a Number.